docker-for-mac/release-notes.md

```diff
@@ -13,14 +13,47 @@ This page contains information about the new features, improvements, known issue
+2021-06-23
 
 [Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
 [Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}
 
 ### New
 
+**Dev Environments Preview**: Dev Environments enable you to seamlessly collaborate with your team members without moving between Git branches to get your code onto your team members' machines. When using Dev Environments, you can share your in-progress work with your team members in just one click, and without having to deal with any merge conflicts. For more information and for instructions on how to use Dev Environments, see [Development Environments Preview](../desktop/dev-environments.md).
+
+### Upgrades
+
+- [Compose V2 beta](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.4)
+  - Fixed a bug where a container cannot be started when a file is bind-mounted into a nested mountpoint. Fixes [docker/compose-cli#1795](https://github.com/docker/compose-cli/issues/1795).
+  - Added support for container links and external links.
+  - Introduced the `docker compose logs --since --until` option.
+  - `docker compose config --profiles` now lists all defined profiles.
+- From [Kubernetes 1.21.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.1) to [Kubernetes 1.21.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.2)
+
+### Bug fixes and minor changes
+- **Volume Management**
+  - Able to remove a file or directory from the Docker Dashboard.
+  - Show more last time modified and size of the contents in the volumes.
+  - The files inside the volume can be saved form the Docker Dashboard.
+- Fixed a bug that caused Docker to fail to start because `/usr/bin` was not present on the `PATH`. Fixes [docker/for-mac#5770](https://github.com/docker/for-mac/issues/5770)
+- Docker Desktop now allows files to be modified inside a host directory which is a nested mountpoint in a container. Fixes [docker/for-mac#5748](https://github.com/docker/for-mac/issues/5748).
+- Fixed a settings migration bug which caused Docker Desktop not to find images and containers data after upgrading to 3.4.0. Fixes [docker/for-mac#5754](https://github.com/docker/for-mac/issues/5754).
+- Fixed a link to the policy that provides details on how Docker handles the uploaded diagnostics data. Fixes [docker/for-mac#5741](https://github.com/docker/for-mac/issues/5741)
+- Docker Desktop now highlights the architecture of the non-native images in the Docker Dashboard on Apple Silicon.
+- Fix using virtualization.framework on macOS 12 (Monterey).
+- The default Unix domain socket name is now `~/.docker/run/docker.sock`. For backwards compatibility `/var/run/docker.sock` is still available. We recommend using the `docker context` called `desktop-linux`.
```

Comment:
I've rearranged the sentences, does this make a bit more sense @stephen-turner @djs55 ?

Reply:
The sentence is written more clearly, but I still don't really know what it means. Why did it change? Is the old name planned to be deprecated? When would I use that docker context?

Reply:
I think it depends on whether the reader is a CLI user or a developer of tools using the Docker API.
- If the user is a CLI user: they don't have to do anything but they will observe the default CLI context is now `desktop-linux`.
- If the user is a developer of Docker API tools: they don't have to do anything either, but they should consider supporting the new socket name `~/.docker/run/docker.sock`

Maybe these should be separated
- The default `docker` CLI `context` is now `desktop-linux`.
- The Docker API is now available on `~/.docker/run/docker.sock`.

(Maybe we don't need to actively recommend anything yet. I wonder whether it would be possible / sensible to add `context` support to the common Docker API clients so that any change in future is transparent. I think the clients already support similar conventions like the `DOCKER_HOST` environment variable. I think of the `docker context` as being a more user-friendly alternative to this variable)